pkg/controller/virtualmachineimport/virtualmachineimport_controller.go

```diff
@@ -170,7 +175,13 @@ func (r *ReconcileVirtualMachineImport) Reconcile(request reconcile.Request) (re
 	if err != nil {
 		return reconcile.Result{}, err
 	}
-
+	mappings, err := prepareResourceMapping(&instance.Spec)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	if err := r.validator.Validate(vm, &request.NamespacedName, mappings); err != nil {
```

Comment:
How do you make sure we validate mappings once and not every cycle of the loop?

Reply:
I don't; the input can change any time - the vm, the mapping resource and the import CR. In the most positive scenario we trigger validation once and it passes. With validation failures we actually want to re-validate in every subsequent iteration.

Reply:
how do we check whether it was positive scenario?

Reply:
The failure is reported as error returned from the `r.validator.Validate` call; if it's nil, we continue with creating VM, otherwise that iteration is stopped and the reconciliation will be executed again, as  the CR is re-queued if error is returned from the `Reconcile` function.

Reply:
I was more thinking about further steps. When we watch dvs or vmi the `Reconcile` is called. How do we make sure that monitoring cycles won't trigger validation again?

Reply:
Oh, OK.
We can potentially base that on the status of the conditions or existence of the vm that is supposed to be created. 

That seems to be a bit wider subject than just validation - what if someone changes the import CR (i.e. network mapping) when the vm exists? We should probably re-validate and update the vm in CNV. If someone changes the storage class, should we do anything with data volume that is currently being transferred to k8s? It looks like o good case for a state machine based on conditions and/or state of sub-processes.

Reply:
In general IMS blocks changes to the mappings during migration. Once VM and DVs are created there is no need to validate or react to changes.

Reply:
If we can assume that the mappings mustn't change after the processing starts, checking if `MappingRulesChecking` completed successfully should be sufficient.